cmake/DyninstPlatform.cmake

```
@@ -47,16 +49,39 @@ if(DYNINST_OS_Linux OR DYNINST_OS_FreeBSD)
   set(DYNINST_OS_UNIX TRUE)
 endif()
 
-if(${_host_arch} IN_LIST _64bit_x86_arches)
-  set(DYNINST_ARCH_x86_64 TRUE)
-elseif(${_host_arch} STREQUAL "aarch64")
-  set(DYNINST_ARCH_aarch64 TRUE)
-elseif(${_host_arch} STREQUAL "ppc64le")
-  set(DYNINST_ARCH_ppc64le TRUE)
-elseif(${_host_arch} IN_LIST _32bit_x86_arches)
-  set(DYNINST_ARCH_i386 TRUE)
+if(DYNINST_CODEGEN_ARCH)
```

To prevent code duplication, let's try

```suggestion
if(DYNINST_CODEGEN_ARCH)
  set(${_host_arch} ${DYNINST_CODEGEN_ARCH})
endif()
```

and then do the checks using `${_host_arch}`.